contractcourt/briefcase.go

```diff
@@ -53,6 +53,28 @@ func (c *ContractResolutions) IsEmpty() bool {
 		c.AnchorResolution == nil && c.BreachResolution == nil
 }
 
+// localForceCloseInfo encapsulates information that led to a channel being
+// force closed with the initiator being local. Currently local force closes
+// can be (1) purely user initiated, (2) automatic due to certain link errors,
+// or (3) automatic due to certain on-chain HTLC conditions. As such, only one
+// of the three fields in this struct will ever have meaningful information. But
+// this may be extended in the future. For example, we may decide to log HTLC
+// information that doesn't automatically result in a force close, but leaves
+// the channel in a state whereby the user has no option but to force close.
+type localForceCloseInfo struct {
+	// userInitiated is true iff the force close was specifically initiated
+	// by the user.
+	userInitiated bool
+
+	// linkFailureError is a non-empty string iff the force close was
+	// automatically initiated following from a link failure.
+	linkFailureError string
+
+	// htlcActions is a non-empty map iff the force close was automatically
+	// initiated by an on-chain trigger such as HTLC timeout.
+	htlcActions ChainActionMap
```

Comment:
I guess it would be nice to be able to correlate the closure resulting from an HTLC with the actual HTLC in the channel (to see why it timed out, what its CLTV delta was and so on).

Reply:
Okay thanks.